### PR TITLE
Pm 9 pets get and post

### PR DESCRIPTION
**Pull request by asching7108**
- Implement GET /pets
- Implement POST /pets
- Change PET.description type from varchar(100) to text in case description is long
- Create sampleDataQueries.sql with shelter and pet sample data for testing

Both tested with Postman and successfully called the GET method from the frontend.
Screen Shot 2022-01-28 at 2 29 13 PM
Screen Shot 2022-01-28 at 2 29 48 PM

### CODE REVIEW

File: database/dataDefinitionQueries.sql
Comment (xHoudek) [inline database/dataDefinitionQueries.sql:36]: Good catch on this! I think we mentioned using longtext, but that's probably overkill to be honest. The text type should be more than enough!

File: src/controllers/petsController.js
Comment (xHoudek) [inline src/controllers/petsController.js]: We should really standardize our indent size. I am a fan of 4 space tabs, but it seems like you may disagree

File: src/models/petsModel.js
Comment (xHoudek) [inline src/models/petsModel.js]: `postPet` is understandable for the controller function as it has to do with the actual endpoint. However, I think these function names should describe the CRUD operation. So this should be `createNewPet`. Admittedly, I should have made my sample function `readAllPets`

File: src/controllers/petsController.js
Comment (xHoudek) [inline src/controllers/petsController.js]: Our controller needs to return the status along with the body. We also need to add error handling in the controllers. Per the Project Plan document on Confluence, the responses for this endpoint are 201 Created, 400 Bad Request, 401 Unauthorized, 406 Not Acceptable

We can probably leave 401 for now, and wait until all the user verification stuff is made

File: src/models/petsModel.js
Comment (xHoudek) [inline src/models/petsModel.js:4]: Thanks for making this an actual query!

File: database/sampleDataQueries.sql
Comment (xHoudek) [inline database/sampleDataQueries.sql:1]: Thanks for adding this file! It's a good idea to have some sample data in the DB

**Review by xHoudek — CHANGES_REQUESTED** [review]
There are some changes needed to this (see comments), but it looks really good! Thank you for adding a bunch of extra (but necessary) stuff!

I don't know what your experience with HTTP error codes is, but if you want help with that just let me know! Otherwise disregard this comment 😄

**Comment by xHoudek** [comment]
I though we agreed on using 2 spaces on Teams. Could you revert back to 2 spaces please?

File: src/models/petsModel.js
Comment (asching7108) [reply src/models/petsModel.js]: Updated in new commits!

File: src/controllers/petsController.js
Comment (asching7108) [reply src/controllers/petsController.js]: - 201 Created - added
- 400 Bad Request - added
- 401 Unauthorized - I added an empty middleware 'requireAuth' in /utils/auth.js and used it for POST /pets. I think it would be cleaner if we could make use of the middleware in server.js for all routes that need it and we don't have to worry about the 401 error anymore in the controllers. You can update the middleware later with the user verification part. Let me know if you have other thoughts!
- 406 Not Acceptable - for POST requests I didn't use any accept headers so I'm not sure when to return this error. I don't think we need any accept header but let me know if you disagree.

**Comment by asching7108** [comment]
> I though we agreed on using 2 spaces on Teams. Could you revert back to 2 spaces please?

Updated!

File: src/utils/auth.js
Comment (xHoudek) [inline src/utils/auth.js]: missing newline

File: test/app.spec.js
Comment (xHoudek) [inline test/app.spec.js]: missing newline

File: src/utils/auth.js
Comment (xHoudek) [inline src/utils/auth.js:2]: Thank you for making this! This should be a good way to ensure requests are authorized

**Review by xHoudek — DISMISSED** [review]
I think changes will need to be made once PM-14 is merged. But as it stand now, it looks good!

**Comment by xHoudek** [comment]
Looks like there are some merge conflicts. Please feel free to merge this once you resolve those conflicts. Thanks!

**Comment by asching7108** [comment]
Hey sorry I tried to solve the conflicts by merging the main branch locally and it dismissed the approval. (Also tried to undo it by the force push and it didn't work). I'm gonna need you to approve it again. Thank you!

**Comment by asching7108** [comment]
Also updated petsController with new input validation methods

**Review by xHoudek — APPROVED** [review]

**Pull request merged**
